Adds new new affordance: placeable and new atomic task: place upright

rebeccazhang0707 opened the pull request:
## Summary
Adds new new affordance: **placeable** and new atomic task: **place upright task**


## Detailed description
- What was the reason for the change?
    - Place upright task is a common inhouse task for humanoids. We believe '**placeable**' should be a basic affordance to support. So we add related atomic task and example to showcase how to use this affordance.
- What has been changed?
    - adds new atomic task: place upright task (and a test script for this task)
    - adss new affordance: placeable (placeable object can be placed upright, like mug, bottle, etc.)
    - adds new example: agibot left_arm place upright mug
    - adds a unitest for 'place_upright_task'
- What is the impact of this change?
    - fixed the bug in agibot.py: since SceneCfg and ActionsCfg cannot accept a new type of property like ArmMode in manager-based workflow


## Test Pipeline
1. Zero Action Policy:
`python isaaclab_arena/examples/policy_runner.py --policy_type zero_action tabletop_place_upright --object mug`

2. Record demos:
`python isaaclab_arena/scripts/imitation_learning/record_demos.py --dataset_file datasets/dataset_agibot_left_arm_rel.hdf5 --num_demos 1 tabletop_place_upright --teleop_device keyboard`

3. Replay demos:
`python isaaclab_arena/scripts/imitation_learning/replay_demos.py --dataset_file datasets/dataset_agibot_left_arm_rel.hdf5 tabletop_place_upright`

4. Test place_upright_mug task:
`pytest isaaclab_arena/tests/test_place_upright_mug.py`

rebeccazhang0707 committed:
Adds new new affordance: placeable and new atomic task: place upright task, and an example of agibot_tabletop_place_upright_mug env

rebeccazhang0707 committed:
adds new TableTopPlaceUprightEnvironment to ExampleEnvironments collection

rebeccazhang0707 committed:
renames MimicArmMode to ArmMode

rebeccazhang0707 committed:
- fixed bugs in agibot.py: SceneCfga nd ActionsCfg cannot be given arm_mode property, which is not a standard scene instance
- refactors placeable's query function

viiik-inside commented on isaaclab_arena/affordances/placeable.py:
```diff
@@ -0,0 +1,187 @@
+            orientation_threshold, device=object_quat.device, dtype=object_quat.dtype
+        )
+
+        success = angle_error < orientation_threshold_tensor
```
nice

rebeccazhang0707 committed:
small fix after rebase on main branch:
- registers retargeter for place_upright_task
- updates background initial_pose and scale

alexmillane commented on isaaclab_arena/affordances/placeable.py:
```diff
@@ -0,0 +1,187 @@
+    - The orientation threshold is the threshold for the angle between the upright axis and the world +Z direction.
+    """
+
+    def __init__(self, upright_axis_name: Literal["x", "y", "z"] = "z", orientation_threshold: float = 0.5, **kwargs):
```
suggestion to add a unit to `orientation_threshold`. For example `orientation_threshold_deg` (if indeed the unit here is degrees.

alexmillane commented on isaaclab_arena/tests/test_place_upright_task.py:
```diff
@@ -0,0 +1,197 @@
+if __name__ == "__main__":
+    # test_place_upright_mug_single()
+    test_place_upright_mug_multi()
+    # test_place_upright_mug_condition()
```
Thanks for adding these tests!

Suggestion to uncommen the lines here.

alexmillane reviewed (approved):
Thanks for adding this. Look good to me!

I have a few comments. Feel free to merge once you have addressed them.

rebeccazhang0707 committed:
refactors placeable affordance, with more clear docstring

viiik-inside committed:
Add add ons to set configurations (#308)

## Summary
We allow addings spawn_cfg_addon and asset_cfg_addon. This exposes and gives possiblitly for a user to create any supported assets with various settings.

rebeccazhang0707 committed:
fixes num_steps in unittest file:
- TODO: find why long steps cause the object behavior not stable
